source/base/CryptoHelper.cpp

```diff
@@ -2,81 +2,23 @@
 
 #include <QCryptographicHash>
 #include <QRegularExpression>
-#include <QDebug>
-
-#include <gcrypt.h>
+#include <QUrl>
 
 const QString CryptoHelper::getHashOfString(const QString& str, bool appendExtention)
 {
     QString hash = QString(QCryptographicHash::hash((str.toUtf8()),QCryptographicHash::Md5).toHex());
 
     if (appendExtention == true)
     {
-        auto index = str.lastIndexOf('.');
-        if (index < str.length() - 1 && index > str.length() - 7) // found . within the last 6 chars (. and 5 extension chars)
+        QString fileName(QUrl(str).fileName());
+
+        auto index = fileName.lastIndexOf('.');
+
+        if (index >=0) // found . 
```

Comment:
This just checks if a . (dot) was found. But this could be anywhere in the string. The original implementation take care of it to be only within the last 6 chars. Not bulled proof, but better then nothing. Please use also this range check.

Reply:
I used the fileName method from QUrl to extract the fileName portion. The extension is the part of the fileName after the last dot. I think this is more secure. It supports any extension length.

Reply:
agree!